hw/ip/flash_ctrl/data/flash_ctrl.sv.tpl

```diff
@@ -775,8 +781,69 @@ module flash_ctrl import flash_ctrl_pkg::*; #(
     .q_o(pwrmgr_o.flash_idle)
   );
 
+  //////////////////////////////////////
+  // Alert senders
+  //////////////////////////////////////
+
+  logic [NumAlerts-1:0] alert_srcs;
+  logic [NumAlerts-1:0] alert_tests;
+
+  logic fatal_err;
+  assign fatal_err = flash_i.flash_alert_p | ~flash_i.flash_alert_n;
+
+  logic recov_mp_err;
+  assign recov_mp_err = flash_mp_error;
+
+  logic recov_ecc_err;
+  assign recov_ecc_err = |flash_i.ecc_single_err | |flash_i.ecc_multi_err;
+
+  assign alert_srcs = { recov_ecc_err,
+                        recov_mp_err,
+                        fatal_err
+                      };
+
+  assign alert_tests = { reg2hw.alert_test.recov_ecc_err.q & reg2hw.alert_test.recov_ecc_err.qe,
+                         reg2hw.alert_test.recov_mp_err.q  & reg2hw.alert_test.recov_mp_err.qe,
+                         reg2hw.alert_test.fatal_err.q     & reg2hw.alert_test.fatal_err.qe
+                       };
+
+  for (genvar i = 0; i < NumAlerts; i++) begin : gen_alert_senders
+    prim_alert_sender #(
+      .AsyncOn(AlertAsyncOn)
```

Comment:
the alert at index 0 is fatal, right?
can you add the following parameter here `.IsFatal(i == 0)`?

Reply:
yeah i need to add that. thanks for the reminder.